libnavigation-base/src/main/java/com/mapbox/navigation/base/formatter/DistanceFormatterOptions.kt

```diff
@@ -73,17 +73,17 @@ class DistanceFormatterOptions private constructor(
      */
     class Builder(applicationContext: Context) {
         private val applicationContext: Context = applicationContext.applicationContext
-        private var unitType: String = VoiceUnit.UNDEFINED
+        private var unitType: UnitType = UnitType.UNDEFINED
```

Comment:
I'd removed `UNDEFINED` at all and take default `UnitType` based on the current location. Below implementation is not relying on `Undefined` and use just default values like(https://github.com/mapbox/mapbox-navigation-android/pull/4224/files#diff-54b35d055593ed228e209be76530c1b732e92a7f2b17a2fd6e00263561f99247L38) :
```kotlin
private val largeUnit = when (options.unitType) {
        UnitType.IMPERIAL -> TurfConstants.UNIT_MILES
        else -> TurfConstants.UNIT_KILOMETERS
    }
```
So if we have `Metrics` or `Undefined` `UnitType` it runs the same way.

Reply:
> So if we have Metrics or Undefined UnitType it runs the same way.

`options` can't have `UNDEFINED` unit type, because when we build `options` we have a check and pass only
`IMPERIAL`/`METRIC`

but I agree, we can remove `UNDEFINED` to make code cleaner.

Reply:
yeah, I saw that :) I just wondering we have `UNDEFINED` as the default one, and it not necessary, as it's replaced by one of relevant, so we can just remove it